server/src/main/java/org/opensearch/index/translog/TranslogWriter.java

```diff
@@ -164,7 +166,8 @@ public static TranslogWriter create(
         final long primaryTerm,
         TragicExceptionHolder tragedy,
         final LongConsumer persistedSequenceNumberConsumer,
-        final BigArrays bigArrays
+        final BigArrays bigArrays,
+        @Nullable IndexSettings indexSettings
```

Comment:
should we just pass the boolean here rather than the indexSettings?

Reply:
should we also try to see if we can pass some predicate (or similar) rather than bring remote store references to TranslogWriter? This is to contain the checks to multiple places. I know we have many presence already. :P

Reply:
done.